network/api/v1/api.go

```diff
@@ -65,11 +66,11 @@ func (a Wrapper) GetTransaction(ctx echo.Context, hashAsString string) error {
 	}
 	transaction, err := a.Service.GetTransaction(hash)
 	if err != nil {
+		if errors.Is(err, dag.ErrTransactionNotFound) {
+			return core.NotFoundError("transaction not found")
```

Comment:
Bit of a vague description. Which transaction are we talking about?

The `err`or is discarded here. What message did it tell?

At this point we have nothing to add to the error (context), and thus err should be returned. GetTransaction(hash) has all the information required to produce a descriptive message. Specifically, it should return something like `core.NetFoundError("hash %#x lookup: %w", hash, ErrNotFound)` with:

```go
var ErrNotFound = errors.New("no such Nuts transaction")
```

Reply:
This is an interpretation of the error to generate the correct response for the API call.
`core.NotFoundError` is used to return a statuscode 404 with the entered message.

Reply:
If it's just a classifier then `core.NotFoundError("%w", err)`.

Reply:
That requires an overhaul of the current API error handling. Out of scope for this PR

Reply:
No just that ☝️ instead of `core.NotFoundError("transaction not found")`.

Reply:
An API should not return an internal error right? Just the fact that the requested item is not found.

Reply:
Correct. APIs should not propagate errors. They log and use status codes.